### PR TITLE
refac: new OutgoingMessage Dto for creating wholesale messages

### PR DESCRIPTION
**Pull request by MadsDue**
## Description
Introduce new WholesaleResultMessageDto for cusomer of the OutgoingMessage module. It represents a wholesale message that is to be sent out. Outgoing Message module will then create 2 outgoing messages for this dto, one for the EnergySupplier and another to the ChargeOwner. 

In the following PRs I will create a specific dtos that you can call in the OutgoingMessage Module with:
- EnergyResultMessageDto
- RequestedEnergyResultAcceptedDto
- RejectedEnergyResultDto

### CODE REVIEW

File: source/OutgoingMessages.Application/OutgoingMessageFactory.cs [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+public class OutgoingMessageFactory
+{
+    private readonly ISystemDateTimeProvider _systemDateTimeProvider;
+
+    public OutgoingMessageFactory(ISystemDateTimeProvider systemDateTimeProvider)
+    {
+        _systemDateTimeProvider = systemDateTimeProvider;
+    }
+
+    public IReadOnlyCollection<OutgoingMessage> CreateMessages(WholesaleResultMessageDto wholesaleResultMessageDto)
```
Comment (lasrinnil) [inline source/OutgoingMessages.Application/OutgoingMessageFactory.cs:34]: A summary describing why multiple messages are made would help the next developer.

File: source/OutgoingMessages.Application/OutgoingMessageFactory.cs [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+                GetChargeOwnerRole(wholesaleResultMessageDto.ChargeOwner),
+                wholesaleResultMessageDto.SenderId,
+                wholesaleResultMessageDto.SenderRole,
+                wholesaleResultMessageDto.SerializedContent,
+                _systemDateTimeProvider.Now(),
+                wholesaleResultMessageDto.RelatedToMessageId),
+        };
+    }
+
+    private static ActorRole GetChargeOwnerRole(ActorNumber chargeOwnerId)
```
Comment (lasrinnil) [inline source/OutgoingMessages.Application/OutgoingMessageFactory.cs:64]: This reads to me much more like domain logic. Than what a factory should be expected to be concerned with.

File: source/IntegrationEvents/IntegrationEvents.Infrastructure/EventProcessors/AmountPerChargeResultProducedV1Processor.cs [hunk 3]
```diff
@@ -17,53 +17,39 @@
...
         _featureManager = featureManager;
-        _actorMessageQueueContext = actorMessageQueueContext;
     }
 
     public string EventTypeToHandle => AmountPerChargeResultProducedV1.EventName;
 
     public async Task ProcessAsync(IntegrationEvent integrationEvent, CancellationToken cancellationToken)
     {
         ArgumentNullException.ThrowIfNull(integrationEvent);
+        if (!await _featureManager.UseAmountPerChargeResultProduced.ConfigureAwait(false))
```
Comment (RasmusGraabaek) [inline source/IntegrationEvents/IntegrationEvents.Infrastructure/EventProcessors/AmountPerChargeResultProducedV1Processor.cs:45]: I would prefer that we have this check, just before we commit?

File: source/OutgoingMessages.Interfaces/IOutgoingMessagesClient.cs [hunk 4]
```diff
@@ -47,4 +47,9 @@ public interface IOutgoingMessagesClient
     ///     Enqueue a message, WITH commit. Currently ONLY used by the Process module wrt reception of events.
     /// </summary>
     Task EnqueueAndCommitAsync(OutgoingMessageDto outgoingMessage, CancellationToken cancellationToken);
+
+    /// <summary>
+    ///     Enqueue wholesale messages based on the dto.
```
Comment (lasrinnil) [inline source/OutgoingMessages.Interfaces/IOutgoingMessagesClient.cs:52]:
```suggestion
    ///     Enqueue wholesale messages. Handles enqueuing messages to all appropriate parties (Receiver, ChargeOwner, TSO).
```

Comment (lasrinnil) [reply source/OutgoingMessages.Interfaces/IOutgoingMessagesClient.cs:52]: Maybe drop the TSO part until implemented

File: source/OutgoingMessages.Interfaces/Models/WholesaleResultMessageDto.cs [hunk 5]
```diff
@@ -17,17 +17,17 @@
...
-public class WholesaleCalculationResultMessage : OutgoingMessageDto
+public class WholesaleResultMessageDto : OutgoingMessageDto
 {
-    private WholesaleCalculationResultMessage(
+    private WholesaleResultMessageDto(
         ActorNumber receiverId,
         Guid processId,
         BusinessReason businessReason,
         ActorRole receiverRole,
+        ActorNumber chargeOwner,
```
Comment (lasrinnil) [inline source/OutgoingMessages.Interfaces/Models/WholesaleResultMessageDto.cs:30]:
```suggestion
        ActorNumber chargeOwnerId,
```

**Review by RasmusGraabaek — DISMISSED** [review]

File: source/OutgoingMessages.Application/OutgoingMessageFactory.cs (hunk 2 above)
Comment (MadsDue) [reply source/OutgoingMessages.Application/OutgoingMessageFactory.cs:64]: I was actaully think about removing the factory again and create a partal class of the OutgoingMessages with this method and a method to create outgoingMessages from the wholesaleResultMessageDto.

File: source/IntegrationEvents/IntegrationEvents.Infrastructure/EventProcessors/AmountPerChargeResultProducedV1Processor.cs (hunk 3 above)
Comment (MadsDue) [reply source/IntegrationEvents/IntegrationEvents.Infrastructure/EventProcessors/AmountPerChargeResultProducedV1Processor.cs:45]: I am not sure the I understand :) can you elaborate?

File: source/OutgoingMessages.Application/Extensions/DependencyInjection/OutgoingMessagesExtensions.cs [hunk 6]
```diff
@@ -53,6 +53,7 @@ public static IServiceCollection AddOutgoingMessagesModule(this IServiceCollecti
         //MessageEnqueueingConfiguration
         services.AddTransient<MessageEnqueuer>()
             .AddScoped<IOutgoingMessageRepository, OutgoingMessageRepository>()
+            .AddTransient<OutgoingMessageFactory>()
```
[the remark this answers is not in the document]
Comment (MadsDue) [reply source/OutgoingMessages.Application/Extensions/DependencyInjection/OutgoingMessagesExtensions.cs:56]: Could argue that both should be a singleton. But that would be in another episode.

**Review by RasmusGraabaek — APPROVED** [review]

**Pull request merged**